frontend/src/metabase/containers/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,184 @@
+
+  if (!error) {
+    setupSingleCollectionEndpoint(COLLECTION_1);
+  } else {
+    fetchMock.get("path:/api/collection/2", { status: 500, body: error });
+  }
+
+  renderWithProviders(
+    <Route
+      path="/"
+      component={() => (
+        <AddToDashSelectDashModal
+          card={card}
+          onChangeLocation={onChangeLocationMocked}
+          onClose={onCloseMocked}
+          dashboards={dashboards}
+        />
+      )}
+    />,
+    { withRouter: true },
+  );
+
+  return {
+    onChangeLocationMocked,
+    onCloseMocked,
+  };
+};
+
+describe("AddToDashSelectDashModal", () => {
+  describe("Create new Dashboard", () => {
+    it("should open CreateDashboardModal", async () => {
+      fetchMock.get("path:/api/collection/1", 200);
+      fetchMock.get("path:/api/collection/2/items", []);
+
+      setup();
+
+      await waitForElementToBeRemoved(() =>
+        screen.queryByTestId("loading-spinner"),
+      );
+
+      const createNewDashboard = screen.getByRole("heading", {
+        name: /create a new dashboard/i,
+      });
+
+      fireEvent.click(createNewDashboard);
+
+      // opened CreateDashboardModal
+      expect(
+        screen.getByRole("heading", {
+          name: /new dashboard/i,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe("Add to existing Dashboard", () => {
+    it("should show loading", () => {
+      fetchMock.get("path:/api/collection/2/items", []);
+      setup();
+
+      expect(screen.getByTestId("loading-spinner")).toBeInTheDocument();
+    });
+
+    it("should show error", async () => {
+      const ERROR = "Server Error!";
+      fetchMock.get("path:/api/collection/2/items", []);
+      setup({ error: ERROR });
+
+      await waitForElementToBeRemoved(() =>
+        screen.queryByTestId("loading-spinner"),
+      );
+
+      expect(screen.getByText(ERROR)).toBeInTheDocument();
+    });
+
+    describe("when user visited some dashboard in last 24hrs", () => {
+      it("should preselected last visited dashboard in the picker", async () => {
+        fetchMock.get("path:/api/collection/2/items", {
+          data: [
+            {
+              name: DASHBOARD.name,
+              id: DASHBOARD.id,
+              model: "dashboard",
+            },
+          ],
+        });
+
+        setup();
+
+        await waitForElementToBeRemoved(() =>
+          screen.queryByTestId("loading-spinner"),
+        );
+
+        // breadcrumbs
+        expect(screen.getByText(COLLECTION_1.name)).toBeInTheDocument();
+        // dashboard item
+        expect(screen.getByText(DASHBOARD.name)).toBeInTheDocument();
+      });
+    });
+
+    describe("when user didn't visit any dashboard during last 24hrs", () => {
+      it("should render root collection without preselection", async () => {
+        fetchMock.get("path:/api/collection/root/items", []);
+
+        setup({
+          dashboard: null,
+        });
+
+        await waitForElementToBeRemoved(() =>
```

Comment:
We also usually add this to `setup`, it should be in the testing docs. In this case `setup` becomes async

Reply:
for both mock `path:/api/collection/root/items` and waitForElementToBeRemoved - if I add them to setup, I'll not be able to test "loading" and "error" state of the component. maybe it's not needed